ksql-cli/src/test/java/io/confluent/ksql/TestRunner.java

```diff
@@ -51,8 +55,16 @@ protected static void test(String command, TestResult.OrderedResult expectedResu
   }
 
   protected static void test(String command, TestResult expectedResult, boolean requireOrder) {
-    TestResult actual = run(command, requireOrder);
-    Assert.assertEquals(expectedResult, actual);
+    run(command, requireOrder);
+    try {
+      TestUtils.waitForCondition(() -> {
+        TestResult actualResult = testTerminal.getTestResult();
+        return actualResult.data.containsAll(expectedResult.data);
+      }, 1000, "Did not get the expected result '" + expectedResult.toString() + ", in a timely fashion.");
+    } catch (InterruptedException e) {
```

Comment:
Where does the interrupted exception come from?

Reply:
From the `TestUtils.waitForCondition` method.

Reply:
ok - i guess i'd just let that bubble out of the test rather than catching it.

Reply:
I debated on this.. This method is used in a lot of places (more than 2 dozen at least), so all of their signatures would have to be updated. I thought this way would create less noise in the code for the same effect: the test would fail.

Reply:
Still, I can change it to bubble up if you think it would be better.

Reply:
is fine